src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -762,6 +762,9 @@ CopyLocalDataIntoShards(Oid distributedRelationId)
 	/* take an ExclusiveLock to block all operations except SELECT */
 	distributedRelation = heap_open(distributedRelationId, ExclusiveLock);
 
+	/* all writes have finished, make sure we can see them */
+	PushActiveSnapshot(GetLatestSnapshot());
```

Comment:
I'd append something like 'we can see them, even when using a transaction snapshot' - It's probably not obvious for the casual reader that GetTransactionSnapshot() would mean something different than GetLatestSnapshot().